runners/core-java/src/test/java/org/apache/beam/runners/core/InMemoryStateInternalsTest.java

```diff
@@ -157,6 +169,70 @@ public void testMergeBagIntoNewNamespace() throws Exception {
   }
 
   @Test
+  public void testSet() throws Exception {
+    SetState<String> value = underTest.state(NAMESPACE_1, STRING_SET_ADDR);
+
+    // State instances are cached, but depend on the namespace.
+    assertEquals(value, underTest.state(NAMESPACE_1, STRING_SET_ADDR));
+    assertFalse(value.equals(underTest.state(NAMESPACE_2, STRING_SET_ADDR)));
+
+    assertThat(value.read(), Matchers.emptyIterable());
+    assertFalse(value.contains("hello"));
+    assertFalse(value.containsAny(Arrays.asList("hello")));
+    value.add("hello");
+    assertThat(value.read(), Matchers.containsInAnyOrder("hello"));
+    assertTrue(value.contains("hello"));
+    assertTrue(value.containsAny(Arrays.asList("hello", "world")));
+    assertFalse(value.containsAll(Arrays.asList("hello", "world")));
+
+    assertFalse(value.addIfAbsent("hello"));
+
+    value.add("world");
+    assertThat(value.read(), Matchers.containsInAnyOrder("hello", "world"));
+    assertTrue(value.containsAll(Arrays.asList("hello", "world")));
+
+    value.clear();
+    assertThat(value.read(), Matchers.emptyIterable());
+    assertThat(underTest.state(NAMESPACE_1, STRING_SET_ADDR), Matchers.sameInstance(value));
+  }
+
```

Comment:
You are still going to add other tests, yes? The other tests in this file should make it easy.